lib/compute_visible_services.ts

```diff
@@ -0,0 +1,165 @@
+  try {
+    const servicesCollectionIterator = await serviceModel.getCollectionIterator();
+    // iterate over the whole services collection and collect visible services
+    const servicesIterator = documentDbUtils.reduceResultIterator(
+      servicesCollectionIterator,
+      (
+        visibleServicesObj: StrMap<VisibleService>,
+        service: RetrievedService
+      ): StrMap<VisibleService> => {
+        // we use an in-memory object (map)
+        // to temporary store visible services
+        const maybeVisibleService = lookup(
+          service.serviceId,
+          visibleServicesObj
+        );
+        if (
+          !service.isVisible &&
+          isSome(maybeVisibleService) &&
+          service.version > maybeVisibleService.value.version
+        ) {
+          // if the service is not visible anymore
+          // delete it from the visible services list
+          winston.debug(
+            "ComputeVisibleService: remove %s",
+            JSON.stringify(maybeVisibleService)
+          );
+          return remove(service.serviceId, visibleServicesObj);
+        } else if (
+          // if the service is visible and we don't have a related
+          // cached visible service yet...
+          service.isVisible &&
+          (isNone(maybeVisibleService) ||
+            // ... or if the updated service is visible and the version
+            // is greater than the stored version of the cached visible service
+            service.version > maybeVisibleService.value.version)
+        ) {
+          winston.debug(
+            "ComputeVisibleService: insert %s",
+            JSON.stringify(service)
+          );
+          // store the visible service
+          return insert(
+            service.serviceId,
+            pick(
+              [
+                "departmentName",
+                "id",
+                "organizationFiscalCode",
+                "organizationName",
+                "serviceId",
+                "serviceName",
+                "version"
+              ],
+              service
+            ),
+            visibleServicesObj
+          );
+        } else {
+          winston.debug(
+            "ComputeVisibleService: noop %s",
+            JSON.stringify(service)
+          );
+        }
+        return visibleServicesObj;
+      }
+    );
+    const errorOrVisibleServices = await documentDbUtils.iteratorToValue(
+      servicesIterator,
+      new StrMap<VisibleService>({})
+    );
+    if (isLeft(errorOrVisibleServices)) {
+      winston.error(
+        "ComputeVisibleServices|Error computing visible services: %s",
+        errorOrVisibleServices.value
+      );
+      return;
+    }
+
+    const visibleServices = errorOrVisibleServices.value;
+    winston.debug(
+      "ComputeVisibleServices|Visible services: %s",
+      JSON.stringify(visibleServices.value)
+    );
+
+    // save to blob storage
+    const errorOrCachedContent = await upsertBlobFromObject(
```

Comment:
can't we use the output bindings here?

Reply:
probably yes, I'll try (I've to check that you can referer to the output storage name from environment variables)